pydicer/input/tcia.py

```diff
@@ -0,0 +1,93 @@
+import os
+import urllib.request, urllib.error, urllib.parse
+import urllib.request, urllib.parse, urllib.error
+from pydicer.input.base import InputBase
+import zipfile
+from pathlib import Path
+
+#
+# Refer https://wiki.cancerimagingarchive.net/display/Public/REST+API+Usage+Guide for complete
+# list of API
+#
+# Some of the code was adapted from the TCIA Python APi client example:
+# https://github.com/TCIA-Community/TCIA-API-SDK/blob/master/tcia-rest-client-python/src/tciaclient.py
+#
+
+
+def run_endpoint(url, query_parameters):
+    """
+    Execute the TCIA API endpoint
+
+    Args:
+        url (str): url of the series to be downloaded and generate its DICOMS
+        query_parameters (dict): dictionary of query parameters to be passed into the api
+        endpoint request.
+
+    Returns:
+        resp (file): the response file in bytes to be written into a zipfile
+    """
+
+    query_parameters = dict((k, v) for k, v in query_parameters.items() if v)
+    queryString = "?%s" % urllib.parse.urlencode(query_parameters)
+    requestUrl = url + queryString
+    request = urllib.request.Request(url=requestUrl)
+    resp = urllib.request.urlopen(request)
+    return resp
+
+
+class TCIAInput(InputBase):
+    def __init__(self, series_instance_uid, working_directory=None):
+        """
+        Input class that interfaces with the TCIA API
+
+        Args:
+            series_instance_uid (str): The series Instance UID to be downloaded from TCIA
+            working_directory (str|pathlib.Path, optional): The working directory in which to
+            store the downloaded data. Defaults to a temp directory
+        """
+        super().__init__(working_directory)
+        self.prefix_url = "https://services.cancerimagingarchive.net/services/v4/TCIA"
+        self.series_instance_uid = series_instance_uid
+
+    def fetch_data(self, output_zip_file="tcia_input_zipfile.zip"):
```

Comment:
I don't think we need to be able to specify the output zip file here do we? Since it seems it is just used temporarily and removed after it is unzipped. Perhaps use tempfile to store it in the system temp directory?

Reply:
Yeah great idea!